src/abstract/PeripheryPayments.sol

```diff
@@ -0,0 +1,70 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+pragma solidity >=0.7.5;
+
+import "@openzeppelin/token/ERC20/IERC20.sol";
+
+import "src/interfaces/IPeripheryPayments.sol";
+import "src/interfaces/IWETH.sol";
+
+import "src/libraries/TransferHelper.sol";
+
+import "src/abstract/PeripheryImmutableState.sol";
+
+abstract contract PeripheryPayments is IPeripheryPayments, PeripheryImmutableState {
+    receive() external payable {
+        require(msg.sender == address(WETH), "Not WETH");
+    }
+
+    /// @inheritdoc IPeripheryPayments
+    function unwrapWETH(uint256 amountMinimum, address recipient) public payable override {
+        uint256 balanceWETH = IWETH(WETH).balanceOf(address(this));
+        require(balanceWETH >= amountMinimum, "Insufficient WETH");
+
+        if (balanceWETH > 0) {
+            IWETH(WETH).withdraw(balanceWETH);
+            TransferHelper.safeTransferETH(recipient, balanceWETH);
+        }
+    }
+
+    /// @inheritdoc IPeripheryPayments
+    function sweepToken(
+        address token,
+        uint256 amountMinimum,
+        address recipient
+    ) public payable override {
+        uint256 balanceToken = IERC20(token).balanceOf(address(this));
+        require(balanceToken >= amountMinimum, "Insufficient token");
+
+        if (balanceToken > 0) {
+            TransferHelper.safeTransfer(token, recipient, balanceToken);
+        }
+    }
+
+    /// @inheritdoc IPeripheryPayments
+    function refundETH() external payable override {
+        if (address(this).balance > 0) TransferHelper.safeTransferETH(msg.sender, address(this).balance);
+    }
```

Comment:
I wonder if we can't implement multicall using function pointers... Something to look at during optimization mode (can you please make this one a github issue). That said, I just checked and delegatecall should only cost as 100 gas (plus or minus the efficiency difference between a JUMP and a *CALL (that resets the stack))